File: src/CourseService/Application/Enrollments/Dtos/EnrolledCourseDto.cs
using Courses.Application.Services.LinkProvider.Abstractions;

namespace Courses.Application.Enrollments.Dtos;

public sealed record EnrolledCourseDto
{
    public required Guid EnrollmentId { get; init; }
    public required Guid CourseId { get; init; }
    public required string CourseTitle { get; init; }
    public required string CourseSlug { get; init; }
    public required double ProgressPercentage { get; init; }
    public required DateTimeOffset? LastAccessedAt { get; init; }
    public required DateTimeOffset EnrolledAt { get; init; }
    public required string Status { get; init; }
    public required IReadOnlyList<LinkDto> Links { get; init; }
}

File: Frontend/src/domain/enrollments/types/EnrolledCourseDto.ts
import type { LinkDto } from "@/shared/types/LinkDto";

/**
 * Backend DTO: Matches EnrolledCourseDto from CourseService (pure, no analytics)
 */
export interface EnrolledCourseDtoApi {
  enrollmentId: string;
  courseId: string;
  courseTitle: string | null;
  courseImageUrl: string | null;
  courseSlug: string | null;
  lastAccessedAt: string | null;
  enrolledAt: string;
  status: string | null;
  links: LinkDto[] | null;
}

/**
 * View model: Flattened shape for components (includes progressPercentage from analytics)
 */
export interface EnrolledCourseDto {
  enrollmentId: string;
  courseId: string;
  courseTitle: string | null;
  courseImageUrl: string | null;
  courseSlug: string | null;
  progressPercentage: number;
  lastAccessedAt: string | null;
  enrolledAt: string;
  status: string | null;
  links: LinkDto[] | null;
}
